repos/system_upgrade/common/libraries/grub.py

```diff
@@ -59,6 +60,22 @@ def get_boot_partition():
     return boot_partition
 
 
+def get_grub_devices():
+    boot_device = get_boot_partition()
+    devices = []
+    if mdraid.is_mdraid_dev(boot_device):
+        component_devs = mdraid.get_component_devices(boot_device)
+        blk_devs = [blk_dev_from_partition(dev) for dev in component_devs]
```

Comment:
Nit: IIUC you can assign directly to `devices`, removing the need for `blk_devs` var completely
```
devices = [blk_dev_from_partition(dev) for dev in component_devs]
```